src/pages/stop/stop.component.ts

```diff
@@ -86,7 +86,15 @@ export class StopComponent {
     this.stopDepartureSvc.getStopDeparture(this.stopId)
       .then(directions => {
         this.sort(directions[0]);
-        this.getRoutes(_.uniq(_.map(directions[0].RouteDirections, 'RouteId')));
+        let routePromises = this.routeSvc.getEachRoute(_.uniq(_.map(directions[0].RouteDirections, 'RouteId')));
+        Promise.all(routePromises).then(routes => {
```

Comment:
Muddling my way through learning a new system: Does this all mean that no routes will get displayed if just one has issues being fetched?

Reply:
I believe so. We had tried appending to the route list as each promise resolved, and then additionally sorting the departures once they had all resolved. For some reason that we couldn't work out, the route list was still empty even after the individual promises had all resolved.

Reply:
Did some undefined handling in https://github.com/umts/pvta-multiplatform/pull/452/commits/18dc68db87974fde2f9278e5f27cacffce838f43, which seems to work. Do you have an idea of how I could test some routes not being fetched?

Reply:
Didn't see this question under outdated, sorry.
What I'd do is add a fake route to the routes you're requesting (hard-code it into your routes-list temporarily), one that you're sure won't have a return. Would that work?

Reply:
Like I'm just thinking, how do I get a 404 normally? I ask for something that doesn't exist.

Reply:
That's a great idea, thank you. It looks like in the case where a 404 is returned, the route service's `response.json() as Route` is then `undefined`. Pushing a bogus route ID to the route ID list doesn't tell us what the display consequences are, though, since there are still no departures with that route ID in the departures list.

So instead I modified the function that determines the query URL, such that it returned something bogus for specifically the 39. What I then saw was this:

<img width="1039" alt="screen shot 2018-02-14 at 1 27 58 pm" src="https://user-images.githubusercontent.com/3988134/36221062-e5e81bd6-118a-11e8-8998-c60fb3daeed9.png">

Reply:
Wait so is that desirable? Is this OK?